File: src/main/java/com/balatro/api/Joker.java
package com.balatro.api;

import com.balatro.api.filter.InBuffonPackFilter;
import com.balatro.enums.JokerType;
import com.balatro.jackson.ItemSerializer;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

@JsonSerialize(using = ItemSerializer.class)
public interface Joker extends Item {

    default Filter inBuffonPack() {
        return new InBuffonPackFilter(this);
    }

    default Filter inBuffonPack(int ante) {
        return new InBuffonPackFilter(ante, this);
    }

    JokerType getType();

    default boolean isRare() {
        return getType() == JokerType.RARE;
    }

    default boolean isCommon() {
        return getType() == JokerType.COMMON;
    }

    default boolean isUncommon() {
        return getType() == JokerType.UNCOMMON;
    }

    default boolean isLegendary() {
        return getType() == JokerType.LEGENDARY;
    }

}

File: docs/js/immolate2/src/class/Joker.js
import { JokerType } from "../enum/JokerType.js";

export class Joker {
	constructor(type, name, eternal = false, prehisable = false, rental = false) {
		this.type = type;
		this.name = name;
		this.eternal = eternal;
		this.prehisable = prehisable;
		this.rental = rental;
	}

	getType() {
		return this.type;
	}

	isRare() {
		return this.type === JokerType.RARE;
	}

	isCommon() {
		return this.type === JokerType.COMMON;
	}

	isUncommon() {
		return this.type === JokerType.UNCOMMON;
	}

	isLegendary() {
		return this.type === JokerType.LEGENDARY;
	}

	getName() {
		return this.name;
	}

	isEternal() {
		return this.eternal;
	}

	isPrehisable() {
		return this.prehisable;
	}

	isRental() {
		return this.rental;
	}

	eq(item) {
		return this.getName() === item.getName();
	}

	equals(value) {
		return this.getName() === value;
	}
}